### PR TITLE
anaconda client integration improvement

### PR DESCRIPTION
**Pull request by kalefranz**
@srossross @dsludwig I'll be hitting you up hopefully sometime today via flowdock, but...

This is a stub PR for functions I need to make anaconda-client work effectively with conda.  Can you help me fill out the code in the stubs?

### CODE REVIEW

File: conda/common/anaconda_client.py [hunk 1]
```diff
@@ -119,23 +119,40 @@
 #     return re.sub(r'([./])conda([./])', r'\1api\2', url, count=1)
 
 
+def get_binstar_directory():
```
Comment (kalefranz) [inline conda/common/anaconda_client.py]: TODO @srossross 

Maybe this is just the binstar_client import.  I don't really care where the binstar directory is or where you're keeping your information.

The real thing I need to do is `read_binstar_tokens` so I can match them up as appropriate.

File: conda/common/anaconda_client.py [hunk 2]
```diff
@@ -119,23 +119,40 @@
...
+    except ImportError:
+        return None
+
+    if 'BINSTAR_CONFIG_DIR' in os.environ:
+        return EnvAppDirs('binstar', 'ContinuumIO', os.environ['BINSTAR_CONFIG_DIR']).user_data_dir
+    else:
+        return AppDirs('binstar', 'ContinuumIO').user_data_dir
+
+
+def get_anaconda_default_site():
```
Comment (kalefranz) [inline conda/common/anaconda_client.py]: TODO @srossross Need pointer to code for this stub.

File: conda/common/anaconda_client.py [hunk 3]
```diff
@@ -119,23 +119,40 @@
...
+    # returns site_name or None if not set
+    pass
+
+
+def get_anaconda_site_url(site_name):
```
Comment (kalefranz) [inline conda/common/anaconda_client.py]: TODO @srossross Need pointer to code for this stub.

File: conda/common/anaconda_client.py [hunk 4]
```diff
@@ -119,23 +119,40 @@
...
+    # returns the url for an anaconda_site
+    pass
+
+
 def replace_first_api_with_conda(url):
     # replace first occurrence of 'api' with 'conda' in url
     return re.sub(r'([./])api([./])', r'\1conda\2', url, count=1)
 
 
 def read_binstar_tokens():
```
Comment (kalefranz) [inline conda/common/anaconda_client.py]: TODO @srossross Need pointer to code that allows me to read currently logged in anaconda.org sites.  Here I'm pulling from the appdirs directory.  I'd rather just use the binstar_client logic for that.

File: conda/common/anaconda_client.py [hunk 5]
```diff
@@ -145,19 +162,13 @@ def read_binstar_tokens():
 
 
 def set_binstar_token(url, token):
```
Comment (kalefranz) [inline conda/common/anaconda_client.py]: TODO @srossross I need this code and `remove_binstar_token`, but only to help effectively test this module.  These two functions are *only* used in tests, and I wouldn't mind moving them to the test_anaconda_client.py module in the future.

**Comment by goanpeca** [comment]
@kalefranz lets create an issue associated to the work in this PR, so we start organizing things better :-)

**Comment by kalefranz** [comment]
Closing for now.  Can reference this PR later for stub work in the future.

**Pull request closed without merging**